### DISCUSSION TITLE
v0.4.4-alpha

### DISCUSSION THREAD

**Discussion by bowring** [CIRDLES/Tripoli#206] [Ideas]
Partial implementation of OGTripoli functionality for IsolinxVersion2 TIMSDP files and Triton exp file.  Try it and if your files won't open, make an issue including an example file.  See our [notes in progress](https://docs.google.com/document/d/14PPEDEJPylNMavpJDpYSuemNb0gF5dz_To3Ek1Y_Agw/edit?usp=sharing).

## What's Changed
* OGTripoli Plotting by @bowring in https://github.com/CIRDLES/Tripoli/pull/203
* Refining plots for OGTripoli case1 (original) by @bowring in https://github.com/CIRDLES/Tripoli/pull/205

**Full Changelog**: https://github.com/CIRDLES/Tripoli/compare/v0.4.3-alpha...v0.4.4-alpha

<hr /><em>This discussion was created from the release <a href='https://github.com/CIRDLES/Tripoli/releases/tag/v0.4.4-alpha'>v0.4.4-alpha</a>.</em>

**Comment by bowring** [comment]
From @noahmclean on  11 Feb 2024:

Feedback so far:
Tripoli has worked on all the .TIMSDP files that I've sent it that were not included in TripoliTestData.  So far so good.
The file import went fine, including choosing ratios and user functions.  
- [x] Maybe instead of or in addition to an invert check-box we can display the inverted ratio?

- [x] The plotting looks good as long as the plots are tiled.  I think 'tiled' is the right option.  There's another option that makes large overlapping windows that I can't get to be useful.

Block vs. Cycle and logRatio vs. Ratio work fine.  
- [x] Will need individual controls, but plots look nice.  
- [ ] Rounding for data reporting still needs tweaks, I'm sure you're waiting on me for some more documentation.
- [x] Is there any way to make the chi look less like an x and more like a greek letter?  There's a unicode chi character out there somewhere that should work in restricted font sets.

- [ ] I'll get you rounding instructions for the reduced chi-square as well.

- [x] Also, the synchronized zoom and pan are not working. 
- [x] For zoom, we shouldn't be able to zoom out past the x-bounds of the analysis.
- [x] Can we use the same zoom/pan mouse gestures here as for the MCMC data?

- [x] The use of open symbols for negative values and closed symbols for positive values doesn't make much sense in most of these plots... user functions, log-ratios particularly.

- [x] I think the 'tile plots' is where everyone will spend their time.  We'll need a better way to figure out how to best tile them -- how many rows and columns should there be?  For instance, if I choose six plots, I get two rows with five columns, so five in the first row and 1 in the second row.

- [ ] I've been working for the last couple days on how to tweak the block weighted means algorithm to get better statistics.  When we take weighted means in Redux, we usually do it on analyses that themselves comprise many points, and so have many (>100, generally) degrees of freedom.  Not so for blocks of discrete data that comprise maybe 10 cycles.
I mocked up some tests where I generated 100 iid random numbers, then calculated the mean and standard error of all 100 or divided them into 10 blocks of 10 ratios each, then used the block weighted mean approach.

- [ ] The block weighted means are just about as close to the true mean as the cycle means are.  But the block weighted mean uncertainties are smaller (and therefore underestimated).  By how much depends on the number of cycles and cyclesPerBlock.  I'll try and come up with a way to adjust.  Likely there will be a fast-and-dirty approach (like an effective degrees of freedom term), and then a more precise way that would look more like our MCMC solutions.  So, there may be a preview/review side to Case1 as well, where you implement a more sophisticated algorithm that runs for a few seconds or minutes, then get back some more sophisticated results.

**Comment by noahmclean** [comment]
Correction to my previous comment: for the reduced chi-square, use a lower case Greek letter chi.
